packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -170,8 +170,8 @@ const VirtualScrollerPinnedRows = styled('div', {
   const boxShadowColor = getBoxShadowColor(theme);
```

Comment:
I am wondering if it'd be better to not show no rows overlay at all even if there are some pinned rows.

```diff
diff --git a/packages/grid/x-data-grid/src/components/base/GridOverlays.tsx b/packages/grid/x-data-grid/src/components/base/GridOverlays.tsx
index 1b1068045..21183218e 100644
--- a/packages/grid/x-data-grid/src/components/base/GridOverlays.tsx
+++ b/packages/grid/x-data-grid/src/components/base/GridOverlays.tsx
@@ -10,6 +10,7 @@ import { gridExpandedRowCountSelector } from '../../hooks/features/filter/gridFi
 import {
   gridRowCountSelector,
   gridRowsLoadingSelector,
+  gridPinnedRowsCountSelector,
 } from '../../hooks/features/rows/gridRowsSelector';
 import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
 import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
@@ -95,10 +96,11 @@ export function GridOverlays() {
   const rootProps = useGridRootProps();
 
   const totalRowCount = useGridSelector(apiRef, gridRowCountSelector);
+  const pinnedRowsCount = useGridSelector(apiRef, gridPinnedRowsCountSelector);
   const visibleRowCount = useGridSelector(apiRef, gridExpandedRowCountSelector);
   const loading = useGridSelector(apiRef, gridRowsLoadingSelector);
 
-  const showNoRowsOverlay = !loading && totalRowCount === 0;
+  const showNoRowsOverlay = !loading && totalRowCount === 0 && pinnedRowsCount === 0;
   const showNoResultsOverlay = !loading && totalRowCount > 0 && visibleRowCount === 0;
 
   let overlay: React.JSX.Element | null = null;
```

Because:

1. No rows not _actually_ true because we do have some rows (even though they are pinned but there are some rows).
2. Even if we assume that we are talking about the unpinned rows, the styling of no rows overlay isn't consistent as the text is always rendered in the middle, e.g compare it with 4 top pinned rows vs 2 top and two bottom pinned rows.
  - 4 top pinned (No rows overlay rendered but hidden behind pinned rows)
   ![image](https://github.com/mui/mui-x/assets/12609561/240386ec-5d5f-4300-bd09-68f3361da481)
  - 2 top pinned and 2 bottom pinned (shows as expected)
   ![image](https://github.com/mui/mui-x/assets/12609561/5cd4f999-8320-4df5-b4ec-0dba098f619d)

What do you think?

Reply:
That can also be an option. I'm ok with either one, to be honest.

Reply:
We could also clarify further by having `No unpinned rows` in that case.